### PR TITLE
SC-7669 frontend datetime library

### PR DESCRIPTION
**Pull request by mwolowyk-cap**
# Description
added datetime library, fixing timezone bugs

## Links to Tickets or other pull requests

- https://ticketsystem.schul-cloud.org/browse/SC-7652
- https://ticketsystem.schul-cloud.org/browse/SC-7667
- https://ticketsystem.schul-cloud.org/browse/SC-7669

## Approval for review
- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

### Link to Definition of Done
More and detailed information on the *definition of done* can be found [on Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

File: controllers/dashboard.js [hunk 1]
```diff
@@ -146,18 +144,16 @@ router.get('/', (req, res, next) => {
...
+							$lte: timesHelper.currentDate().year() + 1,
 						},
 					},
 				],
 			},
 		})
 		.then((data) => data.data.map((homeworks) => {
 			homeworks.secondaryTitle = homeworks.dueDate
-				? moment(homeworks.dueDate).fromNow()
+				? timesHelper.fromUTC(homeworks.dueDate).fromNow()
```
Comment (mwolowyk-cap) [inline controllers/dashboard.js:156]:
```suggestion
				? timesHelper.fromNow(homeworks.dueDate)

File: controllers/dashboard.js [hunk 2]
```diff
@@ -193,7 +189,7 @@ router.get('/', (req, res, next) => {
 		.then((news) => news.data
 			.map((n) => {
 				n.url = `/news/${n._id}`;
-				n.secondaryTitle = moment(n.displayAt).fromNow();
+				n.secondaryTitle = timesHelper.fromUTC(n.displayAt).fromNow();
```
Comment (mwolowyk-cap) [inline controllers/dashboard.js:192]:
```suggestion
				n.secondaryTitle = timesHelper.fromNow(n.displayAt);

**Review by mwolowyk-cap — COMMENTED** [review]
@ABialoskorski please look on comments.

File: controllers/dashboard.js [hunk 3]
```diff
@@ -310,11 +306,11 @@ router.get('/', (req, res, next) => {
 			res.render('dashboard/dashboard', {
 				title: res.$t('dashboard.headline.title'),
 				events: events.reverse(),
-				eventsDate: moment().format('dddd, DD. MMMM YYYY'),
+				eventsDate: timesHelper.currentDate().format(res.$t('format.dateLong')),
```
Comment (mwolowyk-cap) [inline controllers/dashboard.js:309]:
```suggestion
				eventsDate: timesHelper.currentDate().format(timesHelper.FORMAT.dateLong),
```

**Comment by ABialoskorski** [comment]
> @ABialoskorski please look on comments.

done :)

File: controllers/teams.js [hunk 4]
```diff
@@ -292,9 +292,7 @@ router.get('/', async (req, res, next) => {
 		team.background = team.color;
 		team.memberAmount = team.userIds.length;
 		(team.times || []).forEach((time) => {
-			time.startTime = moment(time.startTime, 'x')
-				.utc()
-				.format('HH:mm');
+			time.startTime = timesHelper.fromUTC(time.startTime)
```
[the remark this answers is not in the document]
Comment (mwolowyk-cap) [reply controllers/teams.js:295]: F.e 8:00

File: controllers/teams.js [hunk 5]
```diff
@@ -341,19 +339,19 @@ router.get('/', async (req, res, next) => {
...
 	// map course times to fit model
 	(req.body.times || []).forEach((time) => {
-		time.startTime = moment.duration(time.startTime, 'HH:mm').asMilliseconds();
+		time.startTime = timesHelper.duration(time.startTime, 'HH:mm').asMilliseconds();
 		time.duration = time.duration * 60 * 1000;
 	});
 
 	// eslint-disable-next-line no-underscore-dangle
-	req.body.startDate = moment(req.body.startDate, 'DD:MM:YYYY')._d;
+	req.body.startDate = timesHelper.createFromString(req.body.startDate, timesHelper.FORMAT.date);
```
Comment (mwolowyk-cap) [inline controllers/teams.js:347]: The format is  different DD:MM:YYYY is not equal to FORMAT.date

File: controllers/teams.js [hunk 6]
```diff
@@ -341,19 +339,19 @@ router.get('/', async (req, res, next) => {
...
 	// eslint-disable-next-line no-underscore-dangle
-	req.body.untilDate = moment(req.body.untilDate, 'DD:MM:YYYY')._d;
+	req.body.untilDate = timesHelper.createFromString(req.body.untilDate, timesHelper.FORMAT.date);
 
-	if (!moment(req.body.startDate, 'YYYY-MM-DD').isValid()) {
+	if (!timesHelper.fromUTC(req.body.startDate).isValid()) {
```
Comment (mwolowyk-cap) [inline controllers/teams.js:351]: YYYY-MM-DD is not equal to DD.MM.YYYY

File: controllers/teams.js [hunk 7]
```diff
@@ -341,19 +339,19 @@ router.get('/', async (req, res, next) => {
...
 		delete req.body.startDate;
 	}
-	if (!moment(req.body.untilDate, 'YYYY-MM-DD').isValid()) {
+	if (!timesHelper.fromUTC(req.body.untilDate).isValid()) {
```
Comment (mwolowyk-cap) [inline controllers/teams.js:354]: see above

File: controllers/teams.js [hunk 8]
```diff
@@ -386,14 +384,14 @@ router.post('/copy/:teamId', (req, res, next) => {
...
 
 	// eslint-disable-next-line no-underscore-dangle
-	req.body.startDate = moment(req.body.startDate, 'DD:MM:YYYY')._d;
+	req.body.startDate = timesHelper.createFromString(req.body.startDate, timesHelper.FORMAT.date);
 	// eslint-disable-next-line no-underscore-dangle
-	req.body.untilDate = moment(req.body.untilDate, 'DD:MM:YYYY')._d;
+	req.body.untilDate = timesHelper.createFromString(req.body.untilDate, timesHelper.FORMAT.date);
 
-	if (!moment(req.body.startDate, 'YYYY-MM-DD').isValid()) {
+	if (!timesHelper.fromUTC(req.body.startDate).isValid()) {
```
Comment (mwolowyk-cap) [inline controllers/teams.js:391]: see above

File: controllers/teams.js [hunk 9]
```diff
@@ -736,19 +734,19 @@ router.patch('/:teamId', async (req, res, next) => {
...
 	req.body.times = req.body.times || [];
 	req.body.times.forEach((time) => {
-		time.startTime = moment.duration(time.startTime).asMilliseconds();
+		time.startTime = timesHelper.duration(time.startTime).asMilliseconds();
 		time.duration = time.duration * 60 * 1000;
 	});
 
 	// eslint-disable-next-line no-underscore-dangle
-	req.body.startDate = moment(req.body.startDate, 'DD:MM:YYYY')._d;
+	req.body.startDate = timesHelper.createFromString(req.body.startDate, timesHelper.FORMAT.date);
```
Comment (mwolowyk-cap) [inline controllers/teams.js:742]: format dismatch

**Review by DanielRieckhof — APPROVED** [review]

**Pull request merged**